source/Hardpoint.cpp

```diff
@@ -171,6 +171,15 @@ int Hardpoint::BurstRemaining() const
 
 
 
+// Get the current spinup progress
+double Hardpoint::SpinupProgress() const
+{
+	double spinupTime = outfit->SpinupTime();
+	return spinupTime ? 1. - spinupCount / spinupTime  : 0.;
```

Comment:
I've been struggling with the (seemingly simple) logic of this myself.  `spinupTime` is 0 unless "spinup time" is present in a weapon's stats, so I think you had it right the first time.  Also, you swapped the reload assignments so the fallback value should still be 0, I think.

Reply:
You were right about the `: 1.;` part, but because you introduced the idea of using this comparison means division by 0 is no longer a concern so spinupTime can now remain at 0 and the implicit comparison is now correct.